src/main/java/com/conveyal/datatools/common/utils/SparkUtils.java

```diff
@@ -56,31 +67,73 @@ public static String formatJSON (String key, String value) {
     }
 
     /**
-     * Constructs a JSON string with a result (i.e., OK or ERR), message, code, and if the exception argument is
+     * Constructs an object node with a result (i.e., OK or ERR), message, code, and if the exception argument is
      * supplied details about the exception encountered.
      */
-    public static String formatJSON(String message, int code, Exception e) {
+    public static JsonNode getObjectNode(String message, int code, Exception e) {
         String detail = e != null ? e.getMessage() : null;
         return mapper.createObjectNode()
-                .put("result", code >= 400 ? "ERR" : "OK")
-                .put("message", message)
-                .put("code", code)
-                .put("detail", detail)
-                .toString();
+            .put("result", code >= 400 ? "ERR" : "OK")
+            .put("message", message)
+            .put("code", code)
+            .put("detail", detail);
+    }
+
+    /**
+     * Constructs a JSON string with a result (i.e., OK or ERR), message, code, and if the exception argument is
+     * supplied details about the exception encountered.
+     */
+    public static String formatJSON(String message, int code, Exception e) {
+        return getObjectNode(message, code, e).toString();
+    }
+
+    /**
+     * Wrapper around haltWithMessage for HTTP 500 errors that should be logged and sent to bugsnag
+     */
+    public static void haltWith500(Request request, String message, Exception e) throws HaltException {
+        // log error into datatools log
+        LOG.error(message);
+        e.printStackTrace();
+
+        // create report to notify bugsnag if configured
+        Bugsnag bugsnag = getBugsnag();
+        if (bugsnag != null) {
+            Report report = bugsnag.buildReport(e);
+            Auth0UserProfile userProfile = request.attribute("user");
+            String userEmail = userProfile != null ? userProfile.getEmail() : "no-auth";
+            report.setUserEmail(userEmail);
+            bugsnag.notify(report);
+        }
```

Comment:
What if bugsnag is null? Should we log an error? Also, instead of a `haltWith500` method, I wonder if we should just use a single method (haltWithMessage) and if an error code is 5XX, we perform this bugsnag logic. I worry that future maintainers might just use haltWithMessage and accidentally bypass this method.

Reply:
On second thought, I guess we don't want a missing bugsnag to log an additional warning because this is an optional 3rd party tool. Question still stands for the haltWith500 method though.

Reply:
I like the idea of using only haltWithMessage and sending a bugsnag report if it's setup.